modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/extension/MuleSdkExtensionExtensionModelDeclarer.java

```diff
@@ -38,6 +38,14 @@
 public class MuleSdkExtensionExtensionModelDeclarer {
 
   public static final String EXTENSION_CONSTRUCT_NAME = "extension";
+  public static final String NAME_PARAMETER_NAME = "name";
```

Comment:
shouldn't all these be private?

Reply:
I'm exposing them so that I can use them from the parser, to make sure the same identifiers are used.

Reply:
parser and declarer should not be coupled

Reply:
You are right in that the parser and declarer should not necessarily be coupled. However, in this case, the parser relies on an AST the components of which should match with the declaration of the "extension" extension. We very much want the component identifiers to match the names of the components and constructs of the model. Similarly, we want the component parameter names to match the parameter names in the model. If they don't match, the AST parsing will fail.

For this reason I think having these defined only once makes more sense.

That being said, I like your idea of moving them to a separate constants class. I just pushed that change. Thanks.

Also note that this new class (an interface actually) I put it under `api`, because I suppose these constants will not be allowed to change without breaking backwards. However, we could prefer to leave them under `internal` to be allowed to rename the constants themselves without changing their values. What do you think?

Reply:
should be internal. The class is only a holder for values which reflect an schema. The schema is what's api, not the class holding them

Reply:
Moved to internal now.